cosmos/io.py

```diff
@@ -0,0 +1,183 @@
+from __future__ import annotations
+
+import os
+from pathlib import Path
+from typing import Any
+from urllib.parse import urlparse
+
+from cosmos import settings
+from cosmos.constants import FILE_SCHEME_AIRFLOW_DEFAULT_CONN_ID_MAP
+from cosmos.exceptions import CosmosValueError
+from cosmos.settings import remote_target_path, remote_target_path_conn_id
+
+
+def upload_artifacts_to_aws_s3(project_dir: str, **kwargs: Any) -> None:
```

Comment:
@pankajkoti why haven't we added mandatory args to the function signature? Since we expect those `kwargs` to exist, and we expect end-users to declare them,  it would be clearer to end-users (and us) if those were in the method signature. Example:
```suggestion
def upload_artifacts_to_aws_s3(bucket_name: str, project_dir: str, aws_conn_id: str = S3Hook.default_conn_name, **kwargs: Any) -> None:
```

Reply:
thanks, yes, I was ignorant there. I have added them, but for the default connection IDs I have set them to none since I din't want to have top-level imports for the providers and am initiating those to default in the methods if those connection args are unset.